packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3807,6 +3787,19 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   @pragma('vm:prefer-inline')
   Element inflateWidget(Widget newWidget, Object? newSlot) {
     assert(newWidget != null);
+
+    if (!kReleaseMode && debugProfileBuildsEnabled) {
```

Comment:
are profiles even useful with this much debug code running?

Reply:
In profile mode, this will only add the Widget's name to the timeline _if_ debugProfileBuildsEnabled is manually enabled. (Only in debug mode do you also get extra "timeline arguments" from the diagnostics mode...)

It's sort of the point of the `debugProfileBuildsEnabled` option to get you that extra information in profile mode at the expense of skewing with performance numbers, which is documented on the flag.